docs/content/014-guides/021-abstract-types.md

```diff
@@ -0,0 +1,472 @@
+3. Nexus is smart about what it needs to be satisfied by `__typename` presence. Rather than being a requirement of the model type, it is a requirement of the model type in resolver cases under the union type. For example note below how the `Photo` model type is not required to include `__typename` under the `photos` Query type field:
+
+   ```ts
+   const Query = queryType({
+     definition(t) {
+       t.list.field('photos', {
+         type: 'Photo',
+         resolve(root, args, ctx) {
+           // Nexus does not require __typename here
+           return ctx.db.get.photos()
+         },
+       })
+       t.field('search', {
+         type: 'SearchResult',
+         args: {
+           pattern: stringArg(),
+         },
+         resolve(root, args, ctx) {
+           // Nexus requires __typename here
+           return ctx.db.search(args.pattern).map((result) => {
+             return {
+               ...result,
+               __typename: result.typeName,
+             }
+           })
+         },
+       })
+     },
+   })
+   ```
+
+#### `isTypeOf`
+
+This strategy allows you to discriminate your union member types in a modular (to GraphQL objects) way like `__typename`. Unlike `__typename` it is not based on the shape of your model data. It is instead a predicate function that allows Nexus to know if a given object is of the respective type or not. Here is an example:
+
+```ts
+const Movie = objectType({
+  name: 'Movie',
+  isTypeOf(data) {
+    return Boolean(data.rating)
+  },
+  definition(t) {
+    t.string('url')
+    t.field('rating', {
+      type: 'MovieRating',
+    })
+  },
+})
+
+const Photo = objectType({
+  name: 'Photo',
+  isTypeOf(data) {
+    return Boolean(data.width)
+  },
+  definition(t) {
+    t.string('url')
+    t.int('width')
+    t.int('height')
+  },
+})
+
+const Song = objectType({
+  name: 'Song',
+  isTypeOf(data) {
+    return Boolean(data.album)
+  },
+  definition(t) {
+    t.string('url')
+    t.string('album')
+  },
+})
+```
+
+Like with `resolveType` and `__typename` Nexus leverages TypeScript to ensure your implementation is correct in the following ways:
+
+1. If an object appears in a union type its `isTypeOf` field will be required, unless:
+   - The union type has defined `resolveType`
+   - The model type includes `__typename` property whose type is a string literal matching the GraphQL object name (case sensitive).
+   - The fields where the union type is used include `__typename` in the returned model data
+
+#### Multiple Strategies at Runtime
+
+It is possible to have all three strategies enabled and thus use at once. In that case the following runtime precedence rules apply. Using a strategy here means the implementations of others of lower priority are discarded.
+
+1. `resolveType`
+2. `__typename`
+3. `isTypeOf`
```

Comment:
There no such precedence in the current implementation as we've now moved to the `features` switch. If you enabled all of them, they'll just all error at the same time until you satisfy the requirement of one of the enabled strategy

Reply:
What about a case where multiple strategies are implemented though? Its useful to know what will happen at runtime no?

Reply:
Oh, I see, this is how GraphQL works at runtime, nvm! 👍